tardis/energy_input/gamma_ray_packet_source.py

```diff
@@ -0,0 +1,856 @@
+        energy : float
+            The packet energy
+
+        Returns
+        -------
+        array
+            Array of packet energies
+        """
+        return np.ones(no_of_packets) * energy
+
+    def create_packet_times_uniform_time(self, no_of_packets, start, end):
+        """Samples decay time uniformly (needs non-uniform packet energies)
+
+        Parameters
+        ----------
+        no_of_packets : int
+            Number of packets
+        start : float
+            Start time
+        end : float
+            End time
+
+        Returns
+        -------
+        array
+            Array of packet decay times
+        """
+        z = np.random.random(no_of_packets)
+        decay_times = z * start + (1 - z) * end
+        return decay_times
+
+    def create_packet_times_uniform_energy(
+        self, no_of_packets, isotopes, decay_time
+    ):
+        """Samples the decay time from the mean lifetime of the isotopes
+
+        Parameters
+        ----------
+        no_of_packets : int
+            Number of packets
+        isotopes : pd.Series
+            Series of packet parent isotopes
+        decay_time : array
+            Series of packet decay time index
+
+        Returns
+        -------
+        array
+            Array of decay times
+        """
+        decay_times = np.zeros(len(no_of_packets))
+        for i, isotope in enumerate(isotopes.to_numpy()):
+            decay_time_min = self.times[decay_time[i]]
+            if decay_time_min == self.times[-1]:
+                decay_time_max = self.effective_times[-1]
+            else:
+                decay_time_max = self.times[decay_time[i] + 1]
+            # rejection sampling
+            while (decay_times[i] <= decay_time_min) or (
+                decay_times[i] >= decay_time_max
+            ):
+                decay_times[i] = -self.taus[isotope] * np.log(
+                    np.random.random()
+                )
+        return decay_times
+
+    def create_packets(
+        self, decays_per_isotope, number_of_packets, *args, **kwargs
+    ):
+        """Initialize a collection of GXPacket objects for the simulation
+        to operate on.
+
+        Parameters
+        ----------
+        decays_per_isotope : array int64
+            Probability of decays per simulation shell per isotope per time step
+        number_of_packets : int
+            Number of packets to create
+
+        Returns
+        -------
+        GXPacketCollection
+        """
+        # initialize arrays for most packet properties
+        locations = np.zeros((3, number_of_packets))
+        directions = np.zeros((3, number_of_packets))
+        packet_energies_rf = np.zeros(number_of_packets)
+        packet_energies_cmf = np.zeros(number_of_packets)
+        nus_rf = np.zeros(number_of_packets)
+        nus_cmf = np.zeros(number_of_packets)
+        times = np.zeros(number_of_packets)
+        statuses = np.ones(number_of_packets, dtype=np.int64) * 3
```

Comment:
Why it is multiplied by 3?

Reply:
This is the default status of the packet. I will add a comment to clarify.